include/picongpu/plugins/xrayDiffraction/ReciprocalSpace.hpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2013-2020 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch,
+ *                     Klaus Steiniger, Felix Schmitt, Benjamin Worpitz,
+ *                     Juncheng E, Sergei Bastrakov
+ *                     
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include <pmacc/dimensions/DataSpace.hpp>
+#include <pmacc/dimensions/DataSpaceOperations.hpp>
+
+#include <cstdint>
+
+
+namespace picongpu
+{
+namespace plugins
+{
+namespace xrayDiffraction
+{
+namespace detail
+{
+
+    /** Parameters of the reciprocal space
+     * 
+     * They define the regular 3d Cartesian lattice of scattering vectors q.
+     * The scattering vector here is defined as 4 * pi * sin(theta) / lambda,
+     * where 2 * theta is the angle between scattered and incident beam.
+     */
+    struct ReciprocalSpace
+    {
+        //! Lattice start
+        float3_X min;
+
+        //! Lattice step
+        float3_X step;
+
+        //! Number of points per each direction
+        pmacc::DataSpace< 3 > size;
+
+        /** Create a reciprocal space
+         *
+         * @param min lattice start
+         * @param step lattice step
+         * @param size number of points per each direction
+         */
+        /// TODO: figure out why 0._X instead of 0.0 on Hemera gives
+        /// error #2486: user-defined literal operator not found
+        HDINLINE ReciprocalSpace(
+            float3_X const & min = float3_X::create( 0.0 ),
+            float3_X const & step = float3_X::create( 0.0 ),
```

Comment:
@psychocoderHPC I am confused about the behavior described in the TODO, do you have any idea?

Reply:
Do you tried if the behavior still exists? The PR #2486 is already merged and it looks like #2486 was something different. Could you change it it `0.0_X`

Reply:
You might have linked the wrong PR? The linked one was about the Pi constant.
`_X `: that's exactly the problem, it does not compile on Hemera then, and I am confused why, as described in the TODO in front of the function.

Reply:
I do not know, I have not added the link. It is a new file I have never edited. 

Do you tested `0._X` or `0.0_X` maybe it makes a different.